test/IntegrationTests/TestFixture.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace AskToniApi.IntegrationTests
+{
+    /// <summary>
+    /// A test fixture which hosts the target project (project we wish to test) in an in-memory server.
+    /// </summary>
+    /// <typeparam name="TStartup">Target project's startup type</typeparam>
+    public class TestFixture<TStartup> : IDisposable
+    {
+        private const string SolutionName = "TestingControllersSample.sln";
+        private readonly TestServer _server;
+
+        public TestFixture() : this(Path.Combine("src"))
+        {
+        }
+
+        protected TestFixture(string solutionRelativeTargetProjectParentDir)
+        {
+            var startupAssembly = typeof(TStartup).GetTypeInfo().Assembly;
+            var contentRoot = GetProjectPath(solutionRelativeTargetProjectParentDir, startupAssembly);
+
+            var builder = new WebHostBuilder()
+                .UseContentRoot(contentRoot)
+                .ConfigureServices(InitializeServices)
+                .UseEnvironment("Development")
+                .UseStartup(typeof(TStartup));
+
+            _server = new TestServer(builder);
+
+            Client = _server.CreateClient();
+            Client.BaseAddress = new Uri("http://localhost");
+        }
+
+        public HttpClient Client { get; }
+
+        public void Dispose()
+        {
+            Client.Dispose();
```

Comment:
It's been a while for me, but I think there were some other code you had to call in the dispose method?

Reply:
I don't think so? I didn't actually write this class haha, it's taken directly from https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/testing#integration-testing
So I assume MS knows whats up